File: lib/tiers.ts
import { Tier } from "@prisma/client";

export const TIER_LEVEL: Record<Tier, number> = {
  APPRENTICE: 0,
  JOURNEYMAN: 1,
  MASTER: 2,
  HERO: 3,
};

export const AI_LIMITS: Record<Tier, { messages: number; maxTokens: number }> = {
  APPRENTICE: { messages: 0, maxTokens: 0 },
  JOURNEYMAN: { messages: 15, maxTokens: 600 },
  MASTER: { messages: 100, maxTokens: 1200 },
  HERO: { messages: 300, maxTokens: 2000 },
};

export function canAccess(userTier: Tier, required: Tier) {
  return TIER_LEVEL[userTier] >= TIER_LEVEL[required];
}

File: api/app/core/tiers.py
from app.models.user import TierEnum

AI_LIMITS: dict[TierEnum, dict[str, int]] = {
    TierEnum.APPRENTICE: {"messages": 5, "max_tokens": 400},
    TierEnum.JOURNEYMAN: {"messages": 20, "max_tokens": 600},
    TierEnum.MASTER:     {"messages": 100, "max_tokens": 1200},
    TierEnum.HERO:       {"messages": 300, "max_tokens": 2000},
}

TIER_LEVEL: dict[TierEnum, int] = {
    TierEnum.APPRENTICE: 0,
    TierEnum.JOURNEYMAN: 1,
    TierEnum.MASTER:     2,
    TierEnum.HERO:       3,
}


def can_access(user_tier: TierEnum, required_tier: TierEnum) -> bool:
    return TIER_LEVEL[user_tier] >= TIER_LEVEL[required_tier]


EDUCATIONAL_DISCLAIMER = (
    "This content is provided for educational purposes only "
    "and does not constitute legal or financial advice."
)
